cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/concurrency/ExecutionConcurrencyManager.java

```diff
@@ -131,39 +150,53 @@ public boolean isInitialized() {
   }
 
   /**
-   * Retrieve the movement concurrency of the given concurrency type.
+   * Retrieve the movement concurrency of the given broker concurrency type.
    *
-   * @param brokerId The brokerId to get concurrency
+   * @param brokerId The brokerId to get concurrency.
    * @param concurrencyType The type of concurrency for which the allowed movement concurrency is requested.
    * @return The movement concurrency of the given concurrency type.
    */
-  public int getExecutionConcurrency(int brokerId, ConcurrencyType concurrencyType) {
+  public synchronized int getExecutionBrokerConcurrency(int brokerId, ConcurrencyType concurrencyType) {
     switch (concurrencyType) {
       case INTER_BROKER_REPLICA:
         return interBrokerPartitionMovementConcurrency(brokerId);
       case INTRA_BROKER_REPLICA:
         return intraBrokerPartitionMovementConcurrency(brokerId);
-      case LEADERSHIP:
+      case LEADERSHIP_BROKER:
         return leadershipMovementConcurrency(brokerId);
       default:
         throw new IllegalArgumentException("Unsupported concurrency type " + concurrencyType + " is provided.");
     }
   }
 
+  /**
+   * Retrieve the movement concurrency of the given cluster concurrency type.
+   * @param concurrencyType The type of concurrency for which the allowed movement concurrency is requested.
+   * @return The movement concurrency of the given concurrency type.
+   */
+  public synchronized int getExecutionClusterConcurrency(ConcurrencyType concurrencyType) {
```

Comment:
You don't need to pass in the concurrencyType param here.

Reply:
I think we can make it this way only when we have the need for adding other cluster currency. I would prefer to make it getExecutionClusterLeadershipConcurrency

Reply:
Make sense. Also updated its caller method's name to be more specific for leadership.